onnxruntime/core/providers/stvm/stvm_execution_provider.cc

```diff
@@ -0,0 +1,538 @@
+      }
+
+      tvm::runtime::TVMRetValue rvalue;
+      stvm::TVMRun(*mod_, tensors_outputs_, &rvalue);
+
+      return Status::OK();
+    }
+  private:
+    void getTensorInfo(const TensorShapeProto& shape_proto,
+                       TVMTensorShape& ishape,
+                       size_t indx) {
+      TensorShape ort_shape = utils::GetTensorShapeFromTensorShapeProto(shape_proto);
+      int dims = ort_shape.NumDimensions();
+
+      ishape.resize(dims);
+      for (int j = 0; j < dims; ++j) {
+        int64_t dim = int64_t(ort_shape[j]);
+        if (dim > 0) {
+          ishape[j] = dim;
+        } else {
+          std::cout << "WARNING: input dimension is not positive value (dim = " << dim << "). ";
+          std::cout << "It is replaced by 1. if it needs another value please use provider options to correct it" << std::endl;
+          ishape[j] = 1;
+          update_output_shapes_ = true;
+        }
+      }
+      inputs_info_[indx] = ishape;
+    }
+
+    bool compare_shapes(const TVMTensorShape& shape1, const TVMTensorShape& shape2) {
+      size_t size = shape1.size();
+      if (shape2.size() == size) {
+        for (size_t i = 0; i < size; ++i) {
+          if(shape1[i] != shape2[i]) {
+            return false;
+          }
+        }
+      } else {
+        return false;
+      }
+
+      return true;
+    }
+
+  private:
+    tvm::runtime::Module* mod_;
+    InputsInfoMap inputs_info_;
+    bool update_output_shapes_ = false;
+    TVMTensorShapes output_shapes_;
+    std::vector<DLTensor> tensors_outputs_;
+};
+
+StvmExecutionProvider::StvmExecutionProvider(const StvmExecutionProviderInfo& info)
+    : IExecutionProvider{kStvmExecutionProvider},
+      info_{info} {
+  ProcessInfo();
+
+  AllocatorCreationInfo default_memory_info = {[](int) {
+                                                 return std::make_unique<STVMAllocator>();
+                                               },
+                                               0, false};
+  allocator_ = CreateAllocator(default_memory_info);
+  InsertAllocator(allocator_);
+
+  // Get environment variables
+  const Env& env_instance = Env::Default();
+
+  const std::string dump_subgraphs_env = env_instance.GetEnvironmentVar(stvm_env_vars::kDumpSubgraphs);
+  if (!dump_subgraphs_env.empty()) {
+    dump_subgraphs_ = std::stoi(dump_subgraphs_env) != 0;
+  }
+}
+
+StvmExecutionProvider::~StvmExecutionProvider() {}
+
+AllocatorPtr StvmExecutionProvider::GetAllocator(int id, OrtMemType mem_type) const {
+  return allocator_;
+}
+
+std::vector<std::unique_ptr<ComputeCapability>>
+StvmExecutionProvider::GetCapability(const GraphViewer& graph_viewer,
+                                     const std::vector<const KernelRegistry*>& /*kernel_registries*/) const {
+  std::vector<std::unique_ptr<ComputeCapability>> result;
+  if (graph_viewer.IsSubgraph()) {
+    return result;
+  }
+
+  // Construct modelproto from graph
```

Comment:
Is this creating a copy of the graph/model? I am a bit unclear why.

Reply:
I think you are right, it is deprecated part of code which was created many months ago by my colleagues as a start draft for STVM support and was not rechecked till this moment. Looks that we do not need model proto in GetCapability method, but in Compile only. Therefore I've removed it. Thank you!